Float.Core/UX/CoordinatorParent.cs

```diff
@@ -15,6 +16,11 @@ public abstract class CoordinatorParent : Coordinator, ICoordinatorParent
         /// </summary>
         readonly List<ICoordinator> childCoordinators = new ();
 
+        /// <summary>
+        /// The event args that have been used as we are waiting to finish all the children before closing the parent.
+        /// </summary>
+        EventArgs waitingToFinishEventArgs = null;
+
```

Comment:
Let's make the base implementation `protected` so we only need to define this once.
```suggestion
```